### PR TITLE
rafthttp: remove unncessary go routine

### PR DESCRIPTION
**Pull request by xiang90**
Not sure why there is this additional go routine. So remove it... I am trying to see if there is a performance impact.

/cc @heyitsanthony please give this a careful look. Thanks!

### CODE REVIEW

**Comment by xiang90** [comment]
No real performance gain or loss. Seems like a good code simplification.

File: rafthttp/peer.go [hunk 1]
```diff
@@ -151,66 +146,44 @@ func startPeer(transport *Transport, urls types.URLs, local, to, cid types.ID, r
...
 					plog.Warningf("failed to process raft message (%v)", err)
 				}
-			case urls := <-p.newURLsC:
-				picker.update(urls)
-			case <-p.pausec:
-				paused = true
-			case <-p.resumec:
-				paused = false
 			case <-p.stopc:
 				cancel()
```
Comment (heyitsanthony) [inline rafthttp/peer.go]: If the `propc` `Process` blocks, the close on `stopc` won't be able to cancel the context.

**Comment by heyitsanthony** [comment]
good to see this getting simplified

File: rafthttp/peer.go (hunk 1 above)
Comment (xiang90) [inline rafthttp/peer.go]: right. i will fix.

**Comment by heyitsanthony** [comment]
lgtm

**Pull request merged**